cpp/arcticdb/storage/library_manager.cpp

```diff
@@ -115,14 +119,61 @@ void LibraryManager::write_library_config(const py::object& lib_cfg, const Libra
     );
 }
 
+void LibraryManager::modify_library_option(const arcticdb::storage::LibraryPath &path,
+                                           std::variant<ModifiableLibraryOption, ModifiableEnterpriseLibraryOption> option,
+                                           std::variant<bool, int64_t> new_value) const {
+    // We don't apply a storage override to keep modification backwards compatible.
+    // Before v3.0.0 we didn't use storage overrides and applying one when modifying options would break readers older
+    // than v3.0.0.
+    arcticdb::proto::storage::LibraryConfig config = get_config_internal(path, std::nullopt);
+    auto mutable_write_options = config.mutable_lib_desc()->mutable_version()->mutable_write_options();
+
+    auto get_bool = [&option](std::variant<bool, int64_t> value){
+        util::check(std::holds_alternative<bool>(value), "{} only supports bool values but received {}. Not changing library option.", option, value);
+        return std::get<bool>(value);
+    };
+    auto get_positive_int = [&option](std::variant<bool, int64_t> value){
+        util::check(std::holds_alternative<int64_t>(value) && std::get<int64_t>(value)>0, "{} only supports positive int values but received {}. Not changing library option.", option, value);
+        return std::get<int64_t>(value);
+    };
+
+    util::variant_match(
+            option,
+            [&](const ModifiableLibraryOption& option){
+                switch (option) {
+                    case ModifiableLibraryOption::DEDUP:
```

Comment:
Raise if the option is one that isn't implemented here?

Reply:
Done raises the `UnknownLibraryOption` as before this change.